release-notes/3.10.0.md

```diff
@@ -0,0 +1,577 @@
+
+
+### Separation of tier as its own service
+*Notes for users:*
+This change is to move the management of the tier daemon into the gluster
+service framework, thereby improving it stability and manageability by the
+service framework.
+
+This has no change to any of the tier commands or user facing interfaces and
+operations.
+
+*Limitations:*
+
+*Known Issues:*
+
+### Statedump support for gfapi based applications
+*Notes for users:*
+gfapi based applications now can dump state information for better trouble
+shooting of issues. A statedump can be triggered in two ways:
+
+1. by executing the following on one of the Gluster servers,
+```bash
+   # gluster volume statedump <VOLNAME> client <HOST>:<PID>
+```
+
+   - `<VOLNAME>` should be replaced by the name of the volume
+   - `<HOST>` should be replaced by the hostname of the system running the
+     gfapi application
+   - `<PID>` should be replaced by the PID of the gfapi application
+
+2. through calling `glfs_sysrq(<FS>, GLFS_SYSRQ_STATEDUMP)` within the
+   application
+
+   - `<FS>` should be replaced by a pointer to a `glfs_t` structure
+
+All statedumps (`*.dump.*` files) will be located at the usual location,
+on most distributions this would be `/var/run/gluster/`.
+
+*Limitations:*
+It is not possible to trigger statedumps from the Gluster CLI when the
+gfapi application has lost its management connection to the GlusterD
+servers.
+
+GlusterFS 3.10 is the first release that contains support for the new
+`glfs_sysrq()` function. Applications that include features for
+debugging will need to be adapted to call this function. At the time of
+the release of 3.10, no applications are known to call `glfs_sysrq()`.
+
+*Known Issues:*
+
+### Disabled creation of trash directory by default
+*Notes for users:*
+From now onwards trash directory, namely .trashcan, will not be be created by
+default upon creation of new volumes unless and until the feature is turned ON
+and the restrictions on the same will be applicable as long as features.trash
+is set for a particular volume.
+
+*Limitations:*
+After upgrade for pre-existing volumes, trash directory will be still present at
+root of the volume. Those who are not interested in this feature may have to
+manually delete the directory from the mount point.
+
+*Known Issues:*
+
+### Implemented parallel readdirp with distribute xlator
+*Notes for users:*
+Currently the directory listing gets slower as the number of bricks/nodes
+increases in a volume, though the file/directory numbers remain unchanged.
+With this feature, the performance of directory listing is made mostly
+independent of the number of nodes/bricks in the volume. Thus scale doesn't
+exponentially reduce the directory listing performance. (On a 2, 5, 10, 25 brick
+setup we saw ~5, 100, 400, 450% improvement consecutively)
+
+To enable this feature:
+```bash
+# gluster volume set <VOLNAME> performance.readdir-ahead on
+# gluster volume set <VOLNAME> performance.parallel-readdir on
```

Comment:
Given the sheer improvement possible, I'm surprised this isn't turned on by default.